include/zephyr/drivers/sensor.h

```diff
@@ -1005,7 +1005,50 @@ static inline int sensor_stream(struct rtio_iodev *iodev, struct rtio *ctx, void
 }
 
 /**
- * @brief Read data from a sensor.
+ * @brief Blocking one shot read of samples from a sensor into a buffer
+ *
+ * Using @p cfg, read data from the device by using the provided RTIO context
+ * @p ctx. This call will generate a @ref rtio_sqe that will be given the provided buffer. The call
+ * will wait for the read to complete before returning to the caller.
+ *
+ * @param[in] iodev The iodev created by @ref SENSOR_DT_READ_IODEV
+ * @param[in] ctx The RTIO context to service the read
+ * @param[in] buf Pointer to memory to read sample data into
+ * @param[in] buf_len Size in bytes of the given memory that are valid to read into
+ * @return 0 on success
+ * @return < 0 on error
+ */
+static inline int sensor_read(struct rtio_iodev *iodev, struct rtio *ctx, uint8_t *buf,
```

Comment:
This is synchronous, yet it's in the #ifdef CONFIG_SENSOR_ASYNC_API - intentional?